bigquery_storage/to_dataframe/main.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+
+def main(dataset_id="query_results"):
+    print("Creating clients.")
+    bqclient, bqstorageclient, project_id = create_clients()
+    print("\n\nReading a table:")
+    table_to_dataframe(bqclient, bqstorageclient)
+    print("\n\nReading query results:")
+    query_to_dataframe(bqclient, bqstorageclient, dataset_id)
+    print("\n\nReading a table, using the BQ Storage API directly:")
+    session_to_dataframe(bqstorageclient, project_id)
+
+
+def create_clients():
+    # [START bigquerystorage_pandas_create_client]
+    import google.auth
+    from google.cloud import bigquery
+    from google.cloud import bigquery_storage_v1beta1
+
+    # Explicitly create a credentials object. This allows you to use the same
+    # credentials for both the BigQuery and BigQuery Storage clients, avoiding
+    # unnecessary API calls to fetch duplicate authentication tokens.
+    credentials, project_id = google.auth.default(
+        scopes=["https://www.googleapis.com/auth/cloud-platform"]
+    )
+
+    # Make clients.
+    bqclient = bigquery.Client(credentials=credentials, project=project_id)
+    bqstorageclient = bigquery_storage_v1beta1.BigQueryStorageClient(
+        credentials=credentials
+    )
+    # [END bigquerystorage_pandas_create_client]
+    return bqclient, bqstorageclient, project_id
+
+
+def table_to_dataframe(bqclient, bqstorageclient):
+    from google.cloud import bigquery
```

Comment:
It felt weird to repeat imports multiple times in the same tutorial, but you're right that it's better to show them.

Reply:
Oh it's all for one tutorial? Then I'd just have the imports once in the client initialization snippet. I'd also add "tutorial" somewhere in the region tag name so it's clear they aren't modular samples.

And possibly if you want, you could have a region tag that wraps all the snippets so the user can see what all the code looks like together. That would make this file look kinda weird though because the region tag for the whole tutorial would just have to wrap each of the individual step region tags. Up to you!